src/base/mem_block.hpp

```diff
@@ -70,6 +70,12 @@ namespace mem
 	, ptr { (char *)ptr}
       {}
 
+      virtual ~block()
```

Comment:
Can you please explain the virtual destructor? Do we delete mem::block pointers to derived instances?

Reply:
currently no, but technically it is possible

Reply:
I'd like to suggest hiding the destructor under protected access until we really need it to be virtual.

Reply:
protected: block::~block(); => impossible to use instances of this class outside hierarchy
... and yet it is used in default/private_realloc/dealloc()
There is a workaround for the above particular cases but it is undesirable: to unwrap the code manually each time